torchreid/models/q2l.py

```diff
@@ -0,0 +1,133 @@
+import torch
+import torch.nn as nn
+import math
+
+from .transformer import build_position_encoding
+from .common import ModelInterface
+
+class GroupWiseLinear(nn.Module):
+    # could be changed to:
+    # output = torch.einsum('ijk,zjk->ij', x, self.W)
+    # or output = torch.einsum('ijk,jk->ij', x, self.W[0])
+    def __init__(self, num_class, hidden_dim, use_bias=True):
+        super().__init__()
+        self.num_class = num_class
+        self.hidden_dim = hidden_dim
+        self.use_bias = use_bias
+
+        self.weight = nn.Parameter(torch.Tensor(1, num_class, hidden_dim))
+        if use_bias:
+            self.bias = nn.Parameter(torch.Tensor(1, num_class))
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        stdv = 1. / math.sqrt(self.weight.size(2))
+        for i in range(self.num_class):
+            self.weight[0][i].data.uniform_(-stdv, stdv)
+        if self.use_bias:
+            for i in range(self.num_class):
+                self.bias[0][i].data.uniform_(-stdv, stdv)
+
+    def forward(self, x):
+        # x: B,K,d
+        x = (self.weight * x).sum(-1)
+        if self.use_bias:
+            x = x + self.bias
+        return x
+
+
+class BackboneWrapper(nn.Module):
+    def __init__(self, backbone, position_embedding):
+        super().__init__()
+        self.backbone = backbone
+        self.position_embedding = position_embedding
+        self.num_channels = backbone.num_features
```

Comment:
Since `num_features` is common for all the backbones, I'd rather add abstract method get_num_features() to the model interface.